autoload/lsp/lsp.vim

```diff
@@ -46,13 +46,26 @@ enddef
 # Returns the LSP servers for the a specific filetype. Returns an empty list if
 # the servers is not found.
 def LspGetServers(bnr: number, ftype: string): list<dict<any>>
-  if ftypeServerMap->has_key(ftype)
-    return ftypeServerMap[ftype]
-
-    # TODO score based inclusion/exclusion
+  if !ftypeServerMap->has_key(ftype)
+    return []
   endif
 
-  return []
+  var lspservers: list<dict<any>> = ftypeServerMap[ftype]
+
+  var filteredServers: list<dict<any>> = []
+
+  for lspserver in lspservers
+    # Filter out servers that have had conditional 'runIf' function provided
+    if lspserver->has_key('customRunIf')
+        && lspserver.customRunIf != null_function
+        && !lspserver.customRunIf(bnr)
+      continue
+    endif
+
+    filteredServers->add(lspserver)
+  endfor
+
+  return filteredServers
 enddef
```

Comment:
I have added support for the configuration `runIf` since `rootSearch` cannot be used anymore to determinate if a LSP should be used or not.

Reply:
It will be complicated for a user to write a function (or a lambda) to determine which LSP server should be used.  It is easier for them to specify a list of files or directories to search for to enable a particular LSP server.

Reply:
The only way that we can reuse `rootSearch` is if we change it to include all servers which finds the same root, and exclude all other servers. Otherwise we need to find another name that can be used. `runIfAncestorDirectoriesContains` doesn't feel good 🤔

Reply:
Actually come to think of it, it might be fine in the first iteration to include all LSP servers that have the same score, that will also make us backward compatible.

Reply:
> Actually come to think of it, it might be fine in the first iteration to include all LSP servers that have the same score, that will also make us backward compatible.

I have changed it to that now, and therefore removed `runIf` again.